### PR TITLE
Fix flash attention seed/offset overflow when seed/offset larger than int64 

### PR DESCRIPTION
**Pull request by lixin-sxty**
Operator _scaled_dot_product_flash_attention saves seed and offset when argument dropout is larger than zero for backward gradient calculation.
Torch uses uint64 to represent seed and offset. [See here](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/cuda/detail/PhiloxCudaStateRaw.cuh#L33). However, flash attention uses int64 to save these two data. [See here](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/transformers/cuda/flash_attn/flash_api.cpp#L502-L503)
This will cause an overflow problem when the seed or offset is larger than int64(2^63-1).
Here is a sample for reproducing this bug. Offset has the same problem but it's harder to construct a case.
```
a = torch.randn(1).cuda() # init cuda

dtype = torch.half
shape = (2, 80, 1, 32)
q = torch.randn(shape, dtype=dtype).cuda().requires_grad_()
k = torch.randn(shape, dtype=dtype).cuda().requires_grad_()
v = torch.randn(shape, dtype=dtype).cuda().requires_grad_()

seed = (1 << 64) - 1
print('user seed:', seed)
torch.manual_seed(seed)
out = torch.ops.aten._scaled_dot_product_flash_attention(q, k, v, 0.1)
print('flash attention saved seed for backward:', out[6])

seed = (1 << 63) - 1
torch.manual_seed(seed)
print('user seed:', seed)
out = torch.ops.aten._scaled_dot_product_flash_attention(q, k, v, 0.1)
print('flash attention saved seed for backward:', out[6])

# seed wrong when seed larger than int64
user seed: 18446744073709551615
flash attention saved seed for backward: tensor(-1)
user seed: 9223372036854775807
flash attention saved seed for backward: tensor(9223372036854775807)
```
With this fix, the result is right.

```
# use same code example
# both output are right
user seed: 18446744073709551615
flash attention saved seed for backward: tensor(18446744073709551615, dtype=torch.uint64)
user seed: 9223372036854775807
flash attention saved seed for backward: tensor(9223372036854775807, dtype=torch.uint64)
```

### CODE REVIEW

**Review by drisspg — COMMENTED** [review]
Agree on the change, you will need to change the meta_registrations.py file to fix the return type fake impl

**Comment by lixin-sxty** [comment]
> Agree on the change, you will need to change the meta_registrations.py file to fix the return type fake impl

Ok, I will fix this. By the way, I found that operator _efficient_attention_forward has [the same problem](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/transformers/cuda/attention.cu),  should I fix it in this PR or create another one?

**Comment by drisspg** [comment]
@lixin-sxty feel free to open in new pr

**Comment by lixin-sxty** [comment]
@drisspg Hi, please help review this change. Thanks.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -5307,8 +5307,8 @@ def meta__scaled_dot_product_flash_attention(
         None,
         max_seqlen_batch_q,
         max_seqlen_batch_k,
-        torch.empty((), dtype=torch.long, device="meta"),
```
Comment (drisspg) [inline torch/_meta_registrations.py:5310]: There is also flash_attention_froward that needs to be updated in this file, and then should be good :)

Comment (lixin-sxty) [reply torch/_meta_registrations.py:5310]: Thanks for reminding me, and I have fixed it.

**Review by drisspg — APPROVED** [review]

**Comment by lixin-sxty** [comment]
This PR meets some test case failures in the CI pipeline, and I'm a little bit confused about these failures.

1. https://github.com/pytorch/pytorch/actions/runs/12975226404/job/36334226526?pr=144844 shows a failure in test_slice_with_floordiv;
2. https://github.com/pytorch/pytorch/actions/runs/12975226404/job/36333401251?pr=144844 shows a failure when installing dependency torchtune;
3. https://github.com/pytorch/pytorch/actions/runs/12975226404/job/36333506292?pr=144844 shows a failure in test_second_order_accurate;

These failures seem not related to my changes. Can you help analyze these failures? Thanks. @drisspg

**Comment by drisspg** [comment]
@lixin-sxty I agree, lets rebase and see if they are still present

**Comment by lixin-sxty** [comment]
> @lixin-sxty I agree, lets rebase and see if they are still present

Thanks! All tests passed. Please help merge.

**Comment by lixin-sxty** [comment]
This looks like a CI process issue. What can I do to deal with this?
